src/netaddress.h

```diff
@@ -167,10 +168,13 @@ class CService : public CNetAddr
         template <typename Stream, typename Operation>
         inline void SerializationOp(Stream& s, Operation ser_action) {
             READWRITE(ip);
+
+            // TODO: introduce native support for BE serialization in serialize.h
             unsigned short portN = htons(port);
-            READWRITE(FLATDATA(portN));
-            if (ser_action.ForRead())
+            READWRITE(Span<unsigned char>((unsigned char*)&portN, 2));
```

Comment:
not sure, but maybe use `sizeof(uint16_t)` instead of `2` to increase code readability?
Or use `uint16_t` as htons container (instead of `unsigned short`) and `sizeof(portN)`?

Reply:
Yep, but this code is going away anyhow see 

> @theuni My idea of removing the hack in netaddress is introducing a Big~~Integer~~Endian wrapper (see #10785).